raiden/src/utils/ethers.ts

```diff
@@ -107,3 +107,33 @@ export function getEventsStream<T extends any[]>(
 export async function getNetwork(provider: JsonRpcProvider): Promise<Network> {
   return parseNetwork(parseInt(await provider.send('net_version', [])));
 }
+
+/**
+ * Patch JsonRpcProvider.send to try personal_sign first, and fallback to eth_sign if it fails
+ * Call it once on the provider instance
+ *
+ * @param provider  A JsonRpcProvider instance to patch
+ */
+export function patchSignSend(provider: JsonRpcProvider): void {
+  const origSend: (method: string, params: any) => Promise<any> = provider.send;
+  provider.send = async function(method: string, params: any): Promise<any> {
+    if (method === 'eth_sign') {
+      // try 'personal_sign' by default instead of 'eth_sign'
+      return origSend
+        .bind(this)('personal_sign', [params[1], params[0]])
+        .catch(err => {
+          // on first error, if personal_sign isn't available
+          if (
+            err instanceof Error &&
+            (err.message.includes('The method personal_sign does not exist') ||
+              err.message.includes('Method personal_sign not supported'))
```

Comment:
This 2nd error messages was thrown by `ganache-cli` and caused the e2e tests to fail, so I added it

Reply:
No problem, I suppose we'll need to add other messages here in the future. Error messages aren't very standardized between clients unfortunatelly, and we need to specify each supported cases.